format: HEP detailed record

nikpap opened the pull request:
* Layout and presentation.
* Adds record references.
* Consolidation of common macros between brief and detailed.

Signed-off-by: Nick Papoutsis <admin@nils.gr>

jmartinm commented on inspire/base/static/less/format/detailed-record.less:
```diff
@@ -0,0 +1,279 @@
+ @purple: #293E51;
+
+ #topnav {
+ 	min-height: 65px;
```
here the indentation still looks more than 4 spaces?

nikpap replied:
Is this correct or do I need to edit it? Is there a difference between 4 spaces and 1 tab?

![Indentation]
(http://i.imgur.com/HSjdJup.png)

jmartinm commented on inspire/base/static/less/format/detailed-record.less:
```diff
@@ -0,0 +1,279 @@
+
+	#title {
+		color: grey;
+    	opacity: 0.5;
```
can you amend indentation in all files?

jmartinm commented on inspire/base/static/less/format/detailed-record.less:
```diff
@@ -0,0 +1,279 @@
+ }
+ 
+ body {
+ 	background-color: @teal;
```
So we have a different background color in detailed compared to the rest of the pages?

We already had some non-white background in all pages, how is this one different?

nikpap replied:
This is according to Eamonn's latest design

jmartinm commented on inspire/base/templates/format/record/Inspire_Default_HTML_general_macros.tpl:
```diff
@@ -17,60 +17,156 @@
+  {% set abstract_exists = [] %}
+  {% if record.get('abstract') and record.get('abstract') | is_list %}
+    {% if record.get('abstract')| count > 1%}
+      {% for source in record.get('abstract') %}
```
would be better:

```
{% for abstract in record.get('abstract') %}
```

jmartinm commented on inspire/base/templates/format/record/Inspire_Default_HTML_general_macros.tpl:
```diff
@@ -17,60 +17,156 @@
+          {% do abstract_exists.append(1) %}
+        {% endif %}
+      {% endfor %}
+      {% if abstract %} 
```
Should it be?

```
if abstract_exists
```

jmartinm commented on inspire/base/templates/format/record/Inspire_Default_HTML_general_macros.tpl:
```diff
@@ -17,60 +17,156 @@
+  {% if record.get('report_number') %}
+    {% if record.get('report_number') | is_list() %}
+      {% set filtered_arxiv = record.get('report_number')| remove_duplicates_from_dict() %}
+      {% for i in filtered_arxiv %}
```
Could be better variable name for the for loop.

jmartinm commented on inspire/base/templates/format/record/Inspire_HTML_detailed_macros.tpl:
```diff
@@ -0,0 +1,325 @@
+
+{% from "format/record/Inspire_Default_HTML_general_macros.tpl" import record_cite_modal, record_abstract with context %}
+
+{% macro record_collection_heading() %}
```
This should be part of the breadcrums.

They are automatically generated by the application all over Invenio so we should figure out how to best use them so that all pages are consistent on the place they are shown and how they are displayed.

The first part should actually be a link that goes back to the search results.

For the moment we could leave it like that but we need a different task to convert them into Breadcrums.

jmartinm replied:
This is fine for the moment, we'll see if there is a better dynamic way of defining the breadcrumbs further along.

jmartinm commented on inspire/base/templates/format/record/Inspire_HTML_detailed_macros.tpl:
```diff
@@ -0,0 +1,325 @@
+	</a>
+	{% if record.get('url') %}
+		{% set exists = [] %}
+		{% for url in record.get('url') %}
```
in all this file there is a lot of indentation. Keep with 2 spaces like in the other template files.

nikpap replied:
My tab width is set to 2 for TPL files (and appears correctly on my machine). Might need to check IRL.

jmartinm replied:
It could be tabs VS spaces. Make sure to indent using spaces (2 for template files)

jmartinm commented on inspire/base/templates/format/record/Inspire_HTML_detailed_macros.tpl:
```diff
@@ -0,0 +1,325 @@
+{% endmacro %}
+
+{% macro detailed_record_abstract() %}
+
```
remove extra space ;)

jmartinm commented on inspire/base/templates/format/record/Inspire_HTML_detailed_macros.tpl:
```diff
@@ -0,0 +1,325 @@
+{% endmacro %}
+
+{% macro record_keywords() %}
+<div id="record-keywords">
```
Start with indentation like the other macros

jmartinm commented on inspire/base/templates/format/record/Inspire_HTML_detailed_macros.tpl:
```diff
@@ -0,0 +1,325 @@
+							{% for keywords in record.get('thesaurus_terms') %}
+								{% if 'keyword' in keywords.keys() %}
+								<a href='http://inspirehep.net/search?f=keyword&p="{{ keywords.get('keyword') }}"&ln=en'>{{ keywords.get('keyword') }}</a>,
+								{% endif %}
```
This should search locally (not on `inspirehep`)

jmartinm commented on inspire/base/templates/references.html:
```diff
@@ -0,0 +1,98 @@
+			<span class="reference-citations pull-right">{{ citations }}</span>
+		{% endif %}
+		{% if title %}
+			<div class="reference-title">{{ number }}. <a href="http://inspirehep.net/record/{{recid}}">{{ title }}</a></div>
```
This should be going locally in the final version of this template.

jmartinm commented on inspire/utils/references.py:
```diff
@@ -0,0 +1,77 @@
+            number += 1
+            for reference_field in reference:
+                if 'recid' in reference_field:
+                    recid = reference[reference_field]
```
shouldnt it be"

```python
recid = reference['recid]
```

nikpap replied:
Due to the `if 'recid' in reference_field:`, wouldn't `reference[reference_field]` be the same as ` reference['recid']` ?

jmartinm replied:
Well:

```python
if 'recid' in reference_field
```

Checks if the key `recid` is in the dictionary `reference_field`

Then

```python
reference['recid']
```

would access the value.

If you do:

```python
reference[reference_field]
```

You are accessing the dictionary `reference` with another dictionary `reference_field`. My guess is that the code should break when executed.

jmartinm commented:
Apart from the small comments the code looks good. Did not yet install it in my machine to test.

jalavik commented on inspire/utils/references.py:
```diff
@@ -0,0 +1,77 @@
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2015 CERN.
+##
```
You can remove one of the double `#` in this file to be consistent with the rest.

jalavik commented on inspire/utils/references.py:
```diff
@@ -0,0 +1,77 @@
+
+class Reference(object):
+
+    """docstring for Reference"""
```
Proper docstring would be good here.

jalavik commented on inspire/utils/references.py:
```diff
@@ -0,0 +1,77 @@
+
+    def references(self):
+        """Return reference export for single record."""
+
```
Generally this space can be removed (I believe it breaks a PEP8 rule).

jalavik commented on inspire/utils/references.py:
```diff
@@ -0,0 +1,77 @@
+                                title=title,
+                                authors=authors,
+                                journal=journal)
+                        except Exception:
```
If possible, it is generally safer to specify which type of exception you expect. E.g. `TypeError`, `KeyError` etc. (Several types can be specified like so: `except (TypeError, KeyError)`

jmartinm commented:
Please fix https://travis-ci.org/inspirehep/inspire-next/builds/85348144#L551

jmartinm commented on inspire/base/static/less/format/detailed-record.less:
```diff
@@ -15,8 +16,83 @@
+    padding: 30px 20px 0 20px;
+    z-index: -1;
 
+    .currect-collection {
```
`current` I guess

jmartinm commented on inspire/base/static/less/base/variables.less:
```diff
@@ -21,6 +21,12 @@
 @import (reference) "../inspire.less";
 
+// New design colors
+@blue: #2C3E50;
```
As here you are touching the same file as in https://github.com/inspirehep/inspire-next/pull/418 it would be better if you create a single PR with both commits so that you can fix the conflicts.

jmartinm commented on inspire/base/templates/format/record/Inspire_Default_HTML_general_macros.tpl:
```diff
@@ -17,65 +17,172 @@
+          {% do abstract_exists.append(1) %}
+        {% endif %}
+      {% endfor %}
+      {% for abstract in record.get('abstract') %}
```
The logic of what abstract to display when there are multiple should follow the same rules as in the current inspire. ( see https://github.com/inspirehep/inspire/blob/master/bibformat/format_elements/bfe_INSPIRE_abstract.py#L75-L90 )

* If there are other abstracts apart from the arXiv one, those take precedence. (For example https://inspirehep.net/record/1393642)

I am not sure about the usefulness of showing a second abstract, so for the moment I would show only one.

jmartinm commented on inspire/base/templates/format/record/Inspire_Default_HTML_general_macros.tpl:
```diff
@@ -17,65 +17,172 @@
+    {% set number_of_words = 100 %}
+  {% endif %}
+  {% set abstract_exists = [] %}
+  {% if record.get('abstract') and record.get('abstract') | is_list %}
```
We don't need to do the checks of `is_list` any more as the field should conform with our data model (https://github.com/inspirehep/inspire-next/blob/master/inspire/dojson/hep/schemas/hep-0.0.1.json#L97)

And this will not work as the field is now called `abstracts` instead of `abstract`.

Rebase with latest inspire `next`, reload all the records so that field names are updated and check that the code still works.

jmartinm commented on inspire/base/templates/format/record/Inspire_HTML_detailed_macros.tpl:
```diff
@@ -0,0 +1,327 @@
+
+{% macro record_title() %}
+	{% block title %}
+		{% if record['title'] | is_list %}
```
same here, watch out as the data model has changed https://github.com/inspirehep/inspire-next/blob/master/inspire/dojson/hep/schemas/hep-0.0.1.json#L211

jmartinm commented on inspire/base/templates/format/record/Inspire_HTML_detailed_macros.tpl:
```diff
@@ -0,0 +1,327 @@
+		<i class="fa fa-quote-right"></i> Cite this record
+	</a>
+	{% if record.get('url') %}
+		{% set exists = [] %}
```
The indentation is a bit off compared to the beginning of the file.

jmartinm commented on inspire/base/templates/format/record/Inspire_HTML_detailed_macros.tpl:
```diff
@@ -0,0 +1,327 @@
+{% endmacro %}
+
+{% macro record_doi() %}
+	{% if record.get('doi') %}
```
It is now `dois` https://github.com/inspirehep/inspire-next/blob/master/inspire/dojson/hep/schemas/hep-0.0.1.json#L211

jmartinm commented on inspire/base/templates/format/record/Inspire_HTML_detailed_macros.tpl:
```diff
@@ -0,0 +1,327 @@
+									{% endif %}
+								{% endfor %}
+							</div>
+						</div>
```
Indentation

jmartinm commented on inspire/base/templates/records/Inspire_Default_HTML_detailed_macros.tpl:
```diff
@@ -0,0 +1,176 @@
+      {% set collection_name = collection.get('primary') %}
+      {% if collection_name == "HEP" %}
+        {% if not collection_found %}
+          <span class="currect-collection">Literature</span>
```
Would be good changing to `currect-collection`

jmartinm commented on inspire/base/templates/records/Inspire_Default_HTML_detailed_macros.tpl:
```diff
@@ -0,0 +1,176 @@
+      <span>Conferences</span>
+      <span>Jobs</span>
+    {% endif %}
+  {% endif %}
```
This code can probably be heavily simplified as there is a lot of repetition just to add a class.

Isn't it enough with something like this for each collection?
```
{% if collection_name == "HEP" %}
<span class="currect-collection">Literature</span>
{% else %}
<span>Literature</span>
{% endif %}
```

There might be even be better ways using, for example, string formatting http://jinja.pocoo.org/docs/dev/templates/#format

jmartinm commented on inspire/base/templates/records/Inspire_Default_HTML_detailed_macros.tpl:
```diff
@@ -0,0 +1,176 @@
+          {% set collections = record.collections %}
+          {% for collection in collections %}
+            {% set collection_name = collection.get('primary') %}
+            {% if collection_name == "HEP" %}
```
what about the other collections? They will need the appropriate `cc` parameter as well

jmartinm commented on inspire/base/templates/records/Inspire_Default_HTML_detailed_macros.tpl:
```diff
@@ -0,0 +1,176 @@
+{% endmacro %}
+
+{% macro current_collection() %}
+  {% if record.get('collections') %}
```
All these macros related to collections seem quite similar, is there a way to refactor into a single macro?

jmartinm commented on inspire/base/templates/records/base.html:
```diff
@@ -0,0 +1,83 @@
+
+{% block metaheader %}
+  {% if recid != -1 and config.CFG_WEBSEARCH_DETAILED_META_FORMAT %}
+  {# format_record(record, of=config.CFG_WEBSEARCH_DETAILED_META_FORMAT, ln = g.ln)|safe #}
```
Please, have a look at what is actually being outputted here and if really need it.

jmartinm commented on inspire/base/templates/references.html:
```diff
@@ -0,0 +1,97 @@
+
+<div class="reference-record">
+
+	{% if recid %}
```
Indentation

jmartinm commented:
This commit's message could be improved https://github.com/nikpap/inspire-next/commit/eb668e00dc724209c87bec1a64b148765151d9b9 ;)

jmartinm commented:
Thanks @nikpap , below some comments.

Brief results
-----------------

* All titles are displayed and linked (only one needed)
* `Show 2883 authors & affiliations` No affiliations are shown (and they are not needed in brief results) - if all authors are shown no link is needed.


Detailed view
-----------------

* Duplicate DOIs are not filtered
* The title is too close to the breadcrums, there should be some more padding
* The search bar for Literature has in the HTML
```
<input type="hidden" name="cc" value="CORE">
<input type="hidden" name="cc" value="arXiv">
<input type="hidden" name="cc" value="Citeable">
<input type="hidden" name="cc" value="HEP">
<input type="hidden" name="cc" value="Published">
```
Only:

```
<input type="hidden" name="cc" value="HEP">
```

is needed.

* The last name in the author list has an extra `;`, like `Agatonovic-Jovin, Tatjana ;`
* In the keywords `Show all` would be nice to show the number `Show all 34`
* All the non-link text should not be blue. For example, it was not clear at first sight that these references are not clickable:

![screen shot 2015-10-19 at 5 20 25 pm](https://cloud.githubusercontent.com/assets/580727/10581982/bdff91d2-7685-11e5-8576-c6533f7a0346.png)

* The title should be also black like the rest of the text.
* When loading the detailed page there is the blink of text mode and then the HTML renders
* The scroll bar for references and citations should be limited to the part where the references are (not go up to the filters)
![screen shot 2015-10-19 at 5 23 07 pm](https://cloud.githubusercontent.com/assets/580727/10582053/2cf8949e-7686-11e5-9e7d-ef23f2d980c4.png)
* In my monitor, the placeholder text for the search bar is quite blurry, don't know if it is the usage of Lato the problem

![screen shot 2015-10-19 at 5 24 32 pm](https://cloud.githubusercontent.com/assets/580727/10582089/63f5e906-7686-11e5-8df2-34fe37d2124c.png)
* Show Authors & Affiliations does not show affiliations
* The font size inside the references box might be too big, would be nice to fit more references at first glance.
* References and Citations boxes have no title.

jmartinm commented:
In this commit https://github.com/nikpap/inspire-next/commit/094e12558b3ff6a9735d2caec24c8f90ea30d40c please add spaces between each commit log message.

nikpap commented:
Thanks for the comments @jmartinm 

# Brief results

* https://github.com/inspirehep/inspire-next/pull/429/ PR by @ioannistsanaktsidis seems to fix the brief title issue.
* None of the authors seems to currently have an affiliation - empty field `u'affiliations': []` seems to be the case for all authors in the database. I can recall this used to work as intended in previous versions. Removed display of affiliations in search results.

# Detailed view

* Duplicate DOI's are now filtered
* Title moved further away from breadcrumbs.
* Removed extra `;` after last authors
* Number of keywords now shown next to `Show all`.
* Blink of text while loading eliminated.
* References and citations scroll bar now contained within the center part of the window.
* Search placeholder text has had its font weight reduced and now appears less blurry.
* Affiliations issue mentioned above, in brief results.
* References and citations font size reduced.
* Fixed an issue with references & citations title not displaying.
* Rebased with latest changes on master and updated on github

# Remaining issues

* Non-link text (eg in references, abstract, title) is currently dark blue, based on the mockups. Should that be changed to black?

jmartinm commented:
Regarding affiliations the problem might be a change in the datamodel (`affiliations` instead of `affiliation`).

If I check a record:

```
http://localhost:5000/record/31/export/recjson
```

I can see the affiliations:

```
 {"affiliations": [{"value": "Tel Aviv U."}], "alternative_name": null, "claimed": null, "email": null, "full_name": "Abramowicz, Halina", "inspire_id": "INSPIRE-00060790", "orcid": null, "profile": "", "recid": null, "role": null, "uuid": "a064f07b-ecd3-4559-a7d2-97c2a4704a00"}, {"affiliations": [{"value": "Technion"}],
```

jmartinm commented:
The title in the detailed record still appears quite close and could use some more breathing space.

See

![default](https://cloud.githubusercontent.com/assets/580727/10630765/90cf7bdc-77d9-11e5-8887-36c0ebea8668.png)

VS the mockup

![screen shot 2015-10-21 at 9 53 32 am](https://cloud.githubusercontent.com/assets/580727/10630775/9c95d31c-77d9-11e5-999d-5752fcf4683c.png)

jmartinm commented:
I went to a random record:

```
http://localhost:5000/record/31
```

And the references section does not seem correct (only one shown)

![screen shot 2015-10-21 at 9 55 14 am](https://cloud.githubusercontent.com/assets/580727/10630823/ef532b90-77d9-11e5-8aa1-14de52225710.png)

jmartinm commented:
> Non-link text (eg in references, abstract, title) is currently dark blue, based on the mockups. Should that be changed to black?

I would say change to black. What do you think @eamonnmag ?

nikpap commented:
Follow-up on previous comments by @jmartinm 

* Accommodated the change in the data model regarding affiliations: they now appear correctly.
* Titles in detailed record now appear further from each other - I just realized you meant vertical margin.
* Regarding references: If a reference has a `recid`, all the information is pulled from that record. Currently there are no records associated with each other in the database. Therefore, only unlinked records (missing a `recid`) are currently displayed. Not sure how to further test this with our current dataset. Maybe @kaplun can offer some insight?

jmartinm commented:
Merged

463f426 search: add headers
aca8961 format: detailed record top bars design
47bc5a0 format: fix display of arXiv
7d0a0b4 format: HEP brief search results
f128773 format: HEP detailed record
